src/users/cd-login.vue

```diff
@@ -17,13 +36,52 @@
       return {
         email: '',
         password: '',
+        loginFailed: false,
       };
     },
     methods: {
+      async validateForm() {
+        try {
+          return await this.$validator.validateAll();
+        } catch (e) {
+          return false;
+        }
+      },
       async login() {
-        await UserService.login(this.email, this.password);
-        this.$router.push('/');
+        const valid = await this.validateForm();
+        if (valid) {
+          const response = await UserService.login(this.email, this.password);
+          if (response.body.ok === false) {
+            this.loginFailed = true;
+          } else {
+            this.$router.push(this.$route.query.referer || '/');
```

Comment:
In term of cross-compat, this need to be a refresh, not a router.push as it needs to work with the angular app. That's why you're using a normal <a href> for the "reset" link rather than a <router-link> : it needs to reload the page for angular to load.
Also, i'm pretty sure we have 3 different keyword for referer.. "next" is another one. Now, up to us (cc @DanielBrierton ) to decide if we harmonize it.

Reply:
No, this should be fine. If the route doesn't exist in the Vue app, it will then refresh and defer to the Angular app. If it's in the Vue app, cool, we'll continue.

As for referrer. For starters, I've realised we misspelt refe**rr**er... 😅 Yeah, I wasn't sure if we wanted to settle on just one thing, but obviously, then we'd need to update anywhere in the Angular app where `next` is used, which is a pain. Perhaps we should just support both? Not sure we have anything beyond those two but can check.